galsim/chromatic.py

```diff
@@ -2942,16 +3045,26 @@ def evaluateAtWavelength(self, wave):
                     lam=wave, diam=self.diam,
                     aberrations=self.aberrations*wave_factor, scale_unit=self.scale_unit,
                     _force_stepk=self._stepk*wave_factor, _force_maxk=self._maxk*wave_factor,
-                    gsparams=self.gsparams, **self.kwargs)
+                    geometric_shooting=True, gsparams=self.gsparams, **self.kwargs)
```

Comment:
This is probably fine, as trying to do anything with `geometric_shooting=False` would be really slow.  But I think it's worth mentioning in the docstring that we set this here.  I can imagine someone wanting to see diffraction spikes in their hyper realistic chromatic optics image and being confused when they don't show up.

Reply:
We could make it a class parameter maybe?  But I'm not sure how to write the photon shooting bit in that case.  Do you think it would be feasible to write the `applyTo` function if it's not using the geometric approximation?

Reply:
I think the algorithm is FFT for every wavelength and photon-shoot from that unfortunately.  So possible but immensely slow.  A potential alternative would be to figure out how to enable photon shooting from `InterpolatedChromaticObject` (which aside, I think we mis-named.  Every other chromatic objects starts with `Chromatic` and I keep guessing the wrong name for this one :( ). But I'm not sure what that algorithm should look like...

Reply:
I guess maybe slightly better than the current approximation if people want diffraction spikes would be to still only evaluate it at one wavelength and rescale the x,y values proportional to lambda the way I currently am, but have that base evaluation shoot using geometric_shooting=False.
So then we can have this class have a geometric_shooting option just like the regular OpticalPSF and other PhaseScreenPSFs.  The caveat would be that in this case the chromatic shooting is not expected to be as close a match as when the explicit integral is done.  I don't have a good feel for how big a difference there is in this case.  (The math clearly doesn't have a lambda factor just drop out like it did in the geometric_shooting=True case, but maybe it's not so far off.)

Reply:
Here's a movie I made a few years ago showing the wavelength dependence of an instantaneous atmospheric PSF: https://www.youtube.com/watch?v=x3dTZk5LuGg  (So same math as here, but different wavefront).  It's definitely more complicated than just a simple scaling.

Reply:
Maybe the way to handle this is a combination of what you suggest (scaling by lambda) and an `InterpolatedChromaticObject`.  I.e., draw the FFT images for some small number of control wavelengths, then for a given photon, use its wavelength to pick the nearest FFT image, shoot a photon, and rescale by the difference in wavelengths.  It may even be worth probabilistically selecting between the nearest floor and ceil FFT images to smooth out any kinks between them.

Reply:
OK, I think I have a semi-reasonable implementation of the chromatic shooting when geometric_shooting=False.  It just picks the closest profile among the min, mean, and max wavelength in the photon array, and then rescales the positions by the lambda factor.  
It passes my unit test with the extreme wavelengths, which is kind of a cheat, since all the wavelengths are close to either the min or the max, so it doesn't have to interpolate too far, but I also tested it with a Roman PSF and Roman bandpass, and it works well there too (agreeing at the level of 6e-4).  It's too slow to use as a unit test, but it's in the devel dir.
However, it's still not super fast, since it still needs to do 3 FFTs to make those three profiles and shoot from them.  So I'll work on getting shooting to work with InterpolatedChromaticObject, which if I can get to work, I think will be the preferred way to use ChromaticOpticalPSF with photon shooting.  (It's not yet obvious to me how to get the right interpolated x,y positions out of the saved profiles at the fiducial wavelengths.)